v2/api/network/customizations/virtual_network_extensions.go

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+package customizations
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+
+	network "github.com/Azure/azure-service-operator/v2/api/network/v1alpha1api20201101storage"
+	. "github.com/Azure/azure-service-operator/v2/internal/logging"
+	"github.com/Azure/azure-service-operator/v2/internal/reconcilers/arm"
+	"github.com/Azure/azure-service-operator/v2/internal/resolver"
+	"github.com/Azure/azure-service-operator/v2/internal/util/kubeclient"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/extensions"
+)
+
+var _ extensions.ARMResourceModifier = &VirtualNetworkExtension{}
+
+func (extension *VirtualNetworkExtension) ModifyARMResource(
+	ctx context.Context,
+	armObj genruntime.ARMResource,
+	obj genruntime.MetaObject,
+	kubeClient kubeclient.Client,
+	resolver *resolver.Resolver,
+	log logr.Logger) (genruntime.ARMResource, error) {
+
+	typedObj, ok := obj.(*network.VirtualNetwork)
+	if !ok {
+		return nil, errors.Errorf("cannot run on unknown resource type %T", obj)
```

Comment:
You mean the resource which was not of type `network.VirtualNetwork`? Not directly no, but `%T` of the type will include the Golang package and type name, which is == to the GVK (since the GVK is just identifying the type + package), so I think this has the information we would need to track the problem down.